packages/web/src/components/composer-input/ComposerInput.tsx

```diff
@@ -0,0 +1,363 @@
+          setTimeout(() => {
+            textarea.selectionStart = newCursorPosition
+            textarea.selectionEnd = newCursorPosition
+          }, 0)
+        }
+      }
+    },
+    [isUserAutocompleteActive, onSubmit, handleSubmit, handleBackspace]
+  )
+
+  const handleAutocomplete = useCallback(
+    (user: UserMetadata) => {
+      const cursorPosition = ref.current?.selectionStart || 0
+      const autocompleteRange = isUserAutocompleteActive
+        ? [value.slice(0, cursorPosition).lastIndexOf('@'), cursorPosition]
+        : [0, 1]
+      const mentionText = `@${user.handle}`
+
+      if (!userMentions.includes(mentionText)) {
+        setUserMentions((mentions) => [...mentions, mentionText])
+      }
+      setValue(
+        (value) =>
+          `${value.slice(0, autocompleteRange[0])}${mentionText}${value.slice(
+            autocompleteRange[1]
+          )}`
+      )
+      const textarea = ref.current
+      if (textarea) {
+        setTimeout(() => {
+          textarea.focus()
+          textarea.selectionStart = autocompleteRange[0] + mentionText.length
+          textarea.selectionEnd = autocompleteRange[0] + mentionText.length
+        }, 0)
+      }
+      setIsUserAutocompleteActive(false)
+    },
+    [isUserAutocompleteActive, userMentions, value]
+  )
+
+  const renderChatDisplay = (value: string) => {
+    const cursorPosition = ref.current?.selectionStart || 0
+    const matches = getMatches(value) ?? []
+    const mentions = getUserMentions(value) ?? []
+    const fullMatches = [...matches, ...mentions]
+
+    // If there are no highlightable sections, render text normally
+    if (!fullMatches.length && !isUserAutocompleteActive) {
+      return createTextSections(value)
+    }
+
+    const renderedTextSections = []
+    const autocompleteRange = isUserAutocompleteActive
+      ? [value.slice(0, cursorPosition).lastIndexOf('@'), cursorPosition]
+      : null
+
+    // Filter out matches split by an autocomplete section
+    const filteredMatches = fullMatches.filter(({ index }) => {
+      if (index === undefined) return false
+      if (autocompleteRange) {
+        return !(index >= autocompleteRange[0] && index <= autocompleteRange[1])
+      }
+      return true
+    })
+
+    // Add the autocomplete section
+    if (
+      autocompleteRange &&
+      autocompleteRange[0] >= 0 &&
+      autocompleteRange[1] >= autocompleteRange[0]
+    ) {
+      filteredMatches.push({
+        // TODO: Fix this - type prop issue
+        // @ts-ignore
+        type: 'autocomplete',
+        text: value.slice(autocompleteRange[0], autocompleteRange[1]),
+        index: autocompleteRange[0],
+        link: ''
+      })
+    }
+
+    // Sort match sections by index
+    const sortedMatches = filteredMatches.sort(
+      (a, b) => (a.index ?? 0) - (b.index ?? 0)
+    )
+
+    let lastIndex = 0
+    for (const match of sortedMatches) {
+      // TODO: Fix this - type prop issue
+      // @ts-ignore
```

Comment:
what is the type here that causes this?

Reply:
fixed, was an issue with the type prop, but good now 👍